.github/workflows/raspberry-pi.yaml

```diff
@@ -69,7 +69,43 @@ jobs:
               cd build
 
               cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DBUILD_DEPENDENCIES=OFF -DALIGNED_MEMORY_MODEL=ON
-              make
+              if [ $? -ne 0 ]; then
+                echo "CMake configuration failed!"
+                exit 1
+              fi
+
+              echo "Listing contents of build directory:"
+              ls -l
+          
+              # Running -make- under QEMU inside GitHub Actions has shown instability,
+              # with random segmentation faults occurring sporadically. This could be 
+              # due to QEMU-s emulation overhead, memory constraints, or transient 
+              # issues in the GitHub-hosted environment. To mitigate these failures, 
+              # we retry the build up to 5 times before considering it a hard failure.
+
+              max_retries=5
+              attempt=0
+              while [ $attempt -lt $max_retries ]; do
+                attempt=$((attempt + 1))
+                echo "Attempt $attempt of $max_retries..."
+
+                set +e  # Temporarily disable exit on error
+                make -j
```

Comment:
Might want to consider reverting to just `make` as using more CPU cores to build in parallel might increase resources used which would exacerbate the compiler seg fault problem.

Reply:
Adding to internal tracking